src/unix/core.c

```diff
@@ -983,3 +987,47 @@ int uv__dup2_cloexec(int oldfd, int newfd) {
     return r;
   }
 }
+
+
+int uv_homedir(char* home, size_t* len) {
+
+  char login[LOGIN_NAME_MAX];
+  struct passwd pw;
+  struct passwd* result;
+  char* buf;
+  size_t loginsize = sizeof(login);
+  size_t bufsize = sysconf(_SC_GETPW_R_SIZE_MAX);
+  int r;
+
+  if (home == NULL || len == NULL || *len == 0)
+    return -EINVAL;
+
+  r = getlogin_r(login, loginsize);
+
+  if (r != 0)
+    return -r;
+
+  buf = malloc(bufsize);
+
+  if (buf == NULL)
+    return -ENOMEM;
+
+  r = getpwnam_r((const char*) login, &pw, buf, bufsize, &result);
```

Comment:
You should arguably use `getpwuid_r()` here, then you don't have to call `getlogin_r()` either.  Retry with a bigger buffer when the call fails with errno = ERANGE.

Note that `sysconf(_SC_GETPW_R_SIZE_MAX)` can (at least in theory) fail with -1.  I suggest returning -EIO in that case.